### PR TITLE
Vim eclasses

### PR DESCRIPTION
**Pull request by CyberTailor**

### CODE REVIEW

**Comment by thesamesam** [comment]
Please verify any fixable bugs from https://bugs.gentoo.org/buglist.cgi?quicksearch=vim%20eclass&list_id=6106436 are sorted too.

**Comment by CyberTailor** [comment]
https://bugs.gentoo.org/597304 has been already fixed

**Review by floppym — CHANGES_REQUESTED** [review]
Most of this looks good.

I can't comment on the actual behavior of the src_prepare function in vim-plugin.eclass since I'm unfamiliar how vim plugin tarballs are typically arranged.

File: eclass/vim-plugin.eclass [hunk 1]
```diff
@@ -31,36 +47,95 @@ if [[ ${PV} != 9999* ]] ; then
 fi
 SLOT="0"
 
+if ${_DEFINE_VIM_PLUGIN_SRC_PREPARE}; then
```
[the remark this answers is not in the document]
Comment (CyberTailor) [reply eclass/vim-plugin.eclass:50]: The same check is used in `xdg.eclass`, so I think this is permitted

[the remark this answers is not in the document]
Comment (CyberTailor) [reply eclass/vim-plugin.eclass:51]: I noticed that I forgot to `_DEFINE_VIM_PLUGIN_SRC_PREPARE=false`. Fixed in this PR

[the remark this answers is not in the document]
Comment (thesamesam) [reply eclass/vim-plugin.eclass:51]: Being used elsewhere doesn't mean it's a good pattern. I'd go with what floppym suggested.

[the remark this answers is not in the document]
Comment (CyberTailor) [reply eclass/vim-plugin.eclass]: ok, fixed

**Comment by laumann** [comment]
\o/

**Pull request closed without merging**